src/main/java/Storage.java

```diff
@@ -0,0 +1,132 @@
+package seedu.duke;
+
+import seedu.duke.exception.DukeException;
+import seedu.duke.exception.DukeSaveFailedException;
+import seedu.duke.task.Task;
+import seedu.duke.task.Event;
+import seedu.duke.task.Deadline;
+import seedu.duke.task.Todo;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.BufferedWriter;
+import java.util.ArrayList;
+import java.text.SimpleDateFormat;
+
+/**
+ * Class used to load/save from the save file specified.
+ * @author Lim Daekoon
+ */
+public class Storage {
+
+    private String directory;
+    private FileReader fileReader;
+    private FileWriter fileWriter;
+    private BufferedReader bufferedReader;
+    private BufferedWriter bufferedWriter;
+    private SimpleDateFormat dateFormatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
+
+    /**
+     * Constructs the Storage object with the directory of save file.
+     * @param dir Directory of the save file.
+     */
+    public Storage(String dir) {
+        assert dir == null;
```

Comment:
Shouldn't it be `assert != null` or `assert dir.equals("")` to check if it's empty?